mlkem/ntt.c

```diff
@@ -74,49 +74,95 @@ const int16_t zetas[128] = {
  **************************************************/
 #if !defined(MLKEM_USE_NATIVE_NTT)
 
-// Check that the specific bound for the reference NTT implies
-// the bound required by the C<->Native interface.
-#define NTT_BOUND_REF (5 * MLKEM_Q)
-STATIC_ASSERT(NTT_BOUND_REF <= NTT_BOUND, ntt_ref_bound)
+// Compute a block CT butterflies with a fixed twiddle factor
+STATIC_TESTABLE
+// clang-format off
+void ntt_butterfly_block(int16_t *r, int16_t root, int start, int len, int bound)
+  REQUIRES(0 <= start && start < MLKEM_N && 1 <= len && len <= MLKEM_N / 2 && start + 2 * len <= MLKEM_N)
+  REQUIRES(0 <= bound && bound < INT16_MAX - MLKEM_Q)
+  REQUIRES(IS_FRESH(r, sizeof(int16_t) * MLKEM_N))
+  REQUIRES(SCALAR_ABS_BOUND(root, HALF_Q - 1))
+  REQUIRES(ARRAY_ABS_BOUND(r, 0, start - 1, bound + MLKEM_Q))
+  REQUIRES(ARRAY_ABS_BOUND(r, start, MLKEM_N - 1, bound))
+  ASSIGNS(OBJECT_UPTO(r, sizeof(int16_t) * MLKEM_N))
+  ENSURES(ARRAY_ABS_BOUND(r, 0, start + 2*len - 1, bound + MLKEM_Q))
+  ENSURES(ARRAY_ABS_BOUND(r, start + 2 * len, MLKEM_N - 1, bound))
```

Comment:
and... 

// Remaining coefficients start+2*len through MLKEM_N-1 are unmodified, and therefore retain their original bound

Reply:
I'm unsure. I agree with the comment, but don't want to increase the visual burden of the specification even more. @mkannwischer What do you think?

Reply:
see above